core/model/src/test/java/org/eclipse/rdf4j/model/util/RDFCollectionsTest.java

```diff
@@ -71,35 +70,33 @@ public void testConversionRoundtrip() {
 
 	}
 
-	@Test
-	public void testNonWellformedCollection() {
+	@Test(expected = ModelException.class)
+	public void testNonWellformedCollection_MissingTerminator() {
 		Resource head = vf.createBNode();
 		Model m = RDFCollections.asRDF(values, head, new TreeModel());
 		m.remove(null, RDF.REST, RDF.NIL);
-		try {
-			RDFCollections.asValues(m, head, new ArrayList<>());
-			fail("collection missing terminator should result in error");
-		} catch (ModelException e) {
-			// fall through, expected
-		}
-
-		m = RDFCollections.asRDF(values, head, new TreeModel());
+		RDFCollections.asValues(m, head, new ArrayList<>());
+	}
+
+	@Test(expected = ModelException.class)
+	public void testNonWellformedCollection_Cycle() {
+		Resource head = vf.createBNode("z");
+		Model m = RDFCollections.asRDF(values, head, new TreeModel());
+
+		// Replace rdf:rest relation for head node with one pointing to itself.
+		// This introduces a cycle in an otherwise well-formed collection.
+		m.remove(head, RDF.REST, null);
 		m.add(head, RDF.REST, head);
 
-		try {
-			RDFCollections.asValues(m, head, new ArrayList<>());
-			fail("collection with cycle should result in error");
-		} catch (ModelException e) {
-			// fall through, expected
-		}
-
-		// supply incorrect head node
-		try {
-			RDFCollections.asValues(m, vf.createBNode(), new ArrayList<>());
-			fail("resource that is not a collection should result in error");
-		} catch (ModelException e) {
-			// fall through, expected
-		}
+		RDFCollections.asValues(m, head, new ArrayList<>());
+	}
+
+	@Test(expected = ModelException.class)
+	public void testNonWellformedCollection_IncorrectHeadNode() {
+		Resource head = vf.createBNode();
+		Model m = RDFCollections.asRDF(values, head, new TreeModel());
+
+		RDFCollections.asValues(m, vf.createBNode(), new ArrayList<>());
```

Comment:
Could you add the comment: `// supply incorrect head node` 

Makes the test easier to read.

Reply:
Fixed.